### PR TITLE
TC-DRLK-2.2/2.3/2.12: Modify flow to allow for RequirePINforRemoteOperation to be TRUE or FALSE

### PR DESCRIPTION
**Pull request by jrhees-cae**
TC-DRLK-2.2/2.3/2.12: Modify flow to allow for RequirePINforRemoteOperation to be TRUE or FALSE

- Modify flow to allow for RequirePINforRemoteOperation to be TRUE or FALSE
- Reorder steps 7-9 to ensure subsequent test of WrongCodeEntryLimit doesn't already have existing failed entry attempts
- Fix incorrect PICS (use .Write PICS) for writing WrongCodeEntryLimit and UserCodeTemporaryDisableTime

Fixes https://github.com/CHIP-Specifications/chip-test-plans/issues/3263
Fixes https://github.com/CHIP-Specifications/chip-test-plans/issues/3265
Fixes https://github.com/CHIP-Specifications/chip-test-plans/issues/3266

Accompanying PR for test plan here: https://github.com/CHIP-Specifications/chip-test-plans/pull/3277

### CODE REVIEW

File: scripts/tests/chiptest/__init__.py [hunk 1]
```diff
@@ -150,6 +150,9 @@ def _GetInDevelopmentTests() -> Set[str]:
         "Test_TC_SMOKECO_2_5.yaml",          # chip-repl does not support local timeout (07/20/2023)
         "Test_TC_SMOKECO_2_6.yaml",          # chip-repl does not support local timeout (07/20/2023)
         "Test_TC_IDM_1_2.yaml",              # chip-repl does not support AnyCommands (19/07/2023)
+        "Test_TC_DRLK_2_2.yaml",             # chip-repl does not support EqualityCommands pseudo-cluster (08/04/2023)
+        "Test_TC_DRLK_2_3.yaml",             # chip-repl does not support EqualityCommands pseudo-cluster (08/04/2023)
+        "Test_TC_DRLK_2_12.yaml",            # chip-repl does not support EqualityCommands pseudo-cluster (08/04/2023)
```
Comment (bzbarsky-apple) [inline scripts/tests/chiptest/__init__.py:155]: This is actually disabling this test in CI altogether...  even if you had not removed them from ciTests.json.  Once https://github.com/project-chip/connectedhomeip/pull/28526 merges you should be able to enabled it for chip-tool-python, which is what's wanted here.

File: src/app/tests/suites/ciTests.json [hunk 2]
```diff
@@ -284,15 +284,12 @@
         "DL_LockUnlock",
         "DL_Schedules",
         "Test_TC_DRLK_1_1",
-        "Test_TC_DRLK_2_2",
-        "Test_TC_DRLK_2_3",
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/ciTests.json:288]: Again, after https://github.com/project-chip/connectedhomeip/pull/28526 you could re-enable this in CI.  Maybe a tracking issue?

**Review by bzbarsky-apple — APPROVED** [review]

File: scripts/tests/chiptest/__init__.py (hunk 1 above)
Comment (jrhees-cae) [reply scripts/tests/chiptest/__init__.py:155]: Tracking issue added: https://github.com/project-chip/connectedhomeip/issues/28545

Comment (bzbarsky-apple) [reply scripts/tests/chiptest/__init__.py:155]: @jrhees-cae Note that https://github.com/project-chip/connectedhomeip/pull/28526 is now merged.  Up to you whether you want to re-enable the tests in this PR or a followup.

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
@jrhees-cae it looks like this needs merge with master and regen.

**Pull request merged**